src/test/scripts/check_enterprise_merge.sh

```diff
@@ -1,55 +1,93 @@
 #!/bin/bash
 
+# Testing this script locally requires you to set the following environment
+# variables:
+# CIRCLE_BRANCH, GIT_USERNAME and GIT_TOKEN
+
 # fail if trying to reference a variable that is not set.
 set -u
 # exit immediately if a command fails
 set -e
+# Fail on pipe failures
+set -o pipefail
 
-# try_merge sees if we can merge "from" branch to "to" branch
-# it will exit with nonzero code if the merge fails because of conflicts.
-try_merge() {
-    to=$1
-    from=$2
-    git checkout "${to}"
-    # this will exit since -e option is set and it will return non-zero code on conflicts.
-    git merge --no-ff --no-commit "${from}"
+PR_BRANCH="${CIRCLE_BRANCH}"
+ENTERPRISE_REMOTE="https://${GIT_USERNAME}:${GIT_TOKEN}@github.com/citusdata/citus-enterprise"
 
+# For echo commands "set -x" would show the message effectively twice. Once as
+# part of the echo command shown by "set -x" and once because of the output of
+# the echo command. We do not want "set -x" to show the echo command. We only
+# want to see the actual message in the output of echo itself. This function is
+# a trick to do so. Read the StackOverflow post below to understand why this
+# works and what this works around.
+# Source: https://superuser.com/a/1141026/242593
+shopt -s expand_aliases
+alias echo='{ save_flags="$-"; set +x;} 2> /dev/null; echo_and_restore'
+echo_and_restore() {
+        builtin echo "$*"
+        #shellcheck disable=SC2154
+        case "$save_flags" in
+         (*x*)  set -x
+        esac
 }
 
-cd /tmp
-if [ ! -d citus-enterprise ]; then
-    git clone https://${GIT_USERNAME}:${GIT_TOKEN}@github.com/citusdata/citus-enterprise
+# List executed commands. This is done so debugging this script is easier when
+# it fails. It's explicitely done after git remote add so username and password
+# are not shown in CI output (even though it's also filtered out by CircleCI)
+set -x
+
+# Clone current git repo to a temporary working directory and go there
+GIT_DIR_ROOT="$(git rev-parse --show-toplevel)"
+TMP_GIT_DIR="$(mktemp -d -t citus-merge-check.XXXXXXXXX)"
+git clone "$GIT_DIR_ROOT" "$TMP_GIT_DIR"
+cd "$TMP_GIT_DIR"
+
+# Disable "set -x" again, because $ENTERPRISE_REMOTE contains passwords
+{ set +x ; } 2> /dev/null
```

Comment:
What about we have a function like `sensitive command` which is like the following:

```bash
execute_sensitive_command() {
{ set +x ; } 2> /dev/null
command="$@"
set -x
}

``` 

and then when calling it we can just write:

```bash
execute_sensitive_command git remote add enterprise "$ENTERPRISE_REMOTE"
```

Reply:
I think once there's more than a single one of those commands that might be worth it. Right now not really.